packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -157,6 +157,29 @@ export function navigateBack(params?: object) {
     })
 }
 
+const getActiveRouteState = function (route: NavigationState): NavigationState {
+  if (!route.routes || route.routes.length === 0 || route.index >= route.routes.length) {
+    // TODO: React Navigation types are hard :(
+    // @ts-ignore
+    return route.state
+  }
+
+  const childActiveRoute = (route.routes[route.index] as unknown) as NavigationState
+  return getActiveRouteState(childActiveRoute)
+}
+
+export async function isScreenOnForeground(screen: Screens) {
```

Comment:
I'm not convinced this is the best way of doing this, but I struggled a bit to make React Navigation behave like I wanted it to and this is what I could get working.

Reply:
Looks ok, so checking if the screen was focused didn't work as expected?